Refactor TopologyManagerImport

amichair opened the pull request:
Posting this here for review before pushing, since some of it is a matter of style or preference... I find it much easier to understand now but there may be room for further improvements. What do you think?

Commit:
Refactor TopologyManagerImport

cschneider commented on topology-manager/src/main/java/org/apache/aries/rsa/topologymanager/importer/TopologyManagerImport.java:
```diff
@@ -105,105 +96,86 @@ public void remove(RemoteServiceAdmin rsa) {
         try {
-            unImportForGoneEndpoints(filter);
-            importServices(filter);
+            Set<EndpointDescription> possible = importPossibilities.get(filter);
```
I like the move to streams but I think I would keep the methods unImportForGoneEndpoints(filter) and       importServices(filter) they nicely split up the method.

amichair replied:
Can they be called importAddedEndpoints and unimportRemovedEndpoints or some better name?

cschneider replied:
Sure. I did not want to mandate the naming. Just the idea to separate into two methods.

amichair commented:
Rebased and updated with splitting synchronizeImports as suggested and with the reference null checks from the ARIES-1943 fix.

Note that this does not include the logs that were originally added along with the null checks, as they convolute the code, there is no informative toString in the printed objects, and I'm not sure it's an important case to know about (it's ok if a registration was closed, so as long as there is no NPE bug...). Do you think that's reasonable?

Any other suggestions or improvements?

cschneider commented:
I think it looks good as it is. We can think of further optimisations in the next step if we want.
